### PR TITLE
Support specifying architecture

### PR DESCRIPTION
**Pull request by mit-mit**
Fixes https://github.com/dart-lang/setup-dart/issues/9

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -11,8 +11,12 @@ inputs:
     description: "The channel, or a specific version from a channel to install ('stable', 'beta', 'dev', '2.7.2', '2.12.0-1.4.beta'). Using one of the three channels instead of a version will give you the latest version published to that channel."
     required: false
     default: "stable"
+  architecture:
+    description: "The CPU architecture ('x64 or ia32'). Supported on macOS and Linux only."
```
Comment (athomas) [inline action.yml:15]: Why not Windows?

Comment (athomas) [inline action.yml:15]: Why can you use `arm`/`arm64` on linux?

File: action.yml [hunk 2]
```diff
@@ -11,8 +11,12 @@ inputs:
...
+    required: false
+    default: "x64"
 runs:
   using: "composite"
   steps:
-    - run: $GITHUB_ACTION_PATH/setup.sh ${{ inputs.sdk }} ${{ inputs.channel }} ${{ runner.os }}
+    - run: $GITHUB_ACTION_PATH/setup.sh ${{ inputs.sdk }} ${{ inputs.channel }} ${{ runner.os }} ${{ inputs.architecture }}
```
Comment (athomas) [inline action.yml:21]: You could break up this long line with `\`s.

File: action.yml (hunk 1 above)
Comment (athomas) [reply action.yml:15]: In fact, the test is on windows...

**Review by athomas — APPROVED** [review]

File: action.yml (hunk 1 above)
Comment (mit-mit) [reply action.yml:15]: Hmm, not my a-game when I created this PR. We have ia32 for Windows and Linux, but not macOS, right?

Comment (athomas) [reply action.yml:15]: windows: ia32, x64, (we should probably add arm64 to that list at some point but let's leave that to the Dart PM).
linux: ia32, x64, arm, arm64
mac: x64, (arm64 when we support Apple Silicon natively).

**Comment by mit-mit** [comment]
@athomas can you take a look again?

File: README.md [hunk 3]
```diff
@@ -164,6 +169,10 @@ jobs:
 
 # Version history
 
+## v0.4
+
+  * Added support for specifying the machine architecture (x64 or ia32).
```
Comment (athomas) [inline README.md:174]: Perhaps remove the explicit mention of `x64` and `ia32` here.

Comment (mit-mit) [reply README.md:174]: Yes, one sec!

**Review by athomas — APPROVED** [review]

**Pull request merged**